src/psyclone/tests/psyir/transformations/transformations_test.py

```diff
@@ -538,6 +539,52 @@ def test_omplooptrans_apply(sample_psyir, fortran_writer):
     assert expected in fortran_writer(tree)
 
 
+def test_omploop_trans_new_options(sample_psyir):
+    ''' Thest the new options and validation methods work correctly using
+    OMPLoopTrans apply'''
+    omplooptrans = OMPLoopTrans()
+    tree = sample_psyir.copy()
+
+    # Check we get the relevant error message when adding multiple invalid
+    # options.
+    with pytest.raises(ValueError) as excinfo:
+        omplooptrans.apply(tree.walk(Loop)[0], fakeoption1=1, fakeoption2=2)
+    assert ("'OMPLoopTrans' received invalid options ['fakeoption1', "
+            "'fakeoption2']. Valid options are '['node_type_check', "
+            "'verbose', 'collapse', 'force', 'ignore_dependencies_for', "
+            "'privatise_arrays', 'sequential', 'reprod']."
+            in str(excinfo.value))
+
+    # Check we get the relevant error message when submitting multiple
+    # options with the wrong type
+    with pytest.raises(TypeError) as excinfo:
+        omplooptrans.apply(tree.walk(Loop)[0], verbose=3, force="a")
+    assert ("'OMPLoopTrans' received options with the wrong types:\n"
+            "'verbose' option expects type 'bool' but received '3' "
+            "of type 'int'.\n"
+            "'force' option expects type 'bool' but received 'a' "
+            "of type 'str'.\n"
+            "Please see the documentation and check the provided types."
+            in str(excinfo.value))
+
+    # Check python version, as this tests have different behaviour for
+    # new python versions vs 3.8 or 3.7.
```

Comment:
Please add a TODO here to the appropriate issue if we can remove this bit once we drop support for 3.8.

Reply:
Added - i think its the right issue (the only one I know of regarding 3.7/3.8 removal).